### ISSUE TITLE
GRE Tunnel Declaration and object reference support

### ISSUE THREAD

**Issue by steveh565** [F5Networks/f5-declarative-onboarding#98]
### Is your feature request related to a problem? Please describe.
I need to be able to declare a GRE Tunnel configuration.
I need to be able to declare a route that references the GRE Tunnel
I need to be able to declare a self-ip that listens on the GRE tunnel

### Describe the solution you'd like

I would like to be able to include declarations for a GRE tunnel and reference it in Virtual Server, Self-IP and Route declarations, as follows:
```
    "greTunnel1": {
        "class": "Tunnel",
        "type": "gre",
        "localAddr": "10.90.4.8",
        "remoteAddr": "10.190.2.4",
        "mode": "bidirectional",
        "mtu": "0",
        "usePMTU": true,
        "isTransparent": false
    },
    "greTunnel1-self": {
       "class": "SelfIp",
       "address": "10.90.4.8/24",
       "vlan": "greTunnel1",
       "allowService": "default",
       "trafficGroup": "traffic-group-local-only"
   },
   "greTunnel1-traffic-route": {
       "class": "Route",
       "gw": "greTunnel1",
       "network": "10.190.100.0/24",
       "mtu": 1500
   }
```

### Describe alternatives you've considered

scripting tmsh commands  :(

**Comment by dstokesf5** [comment]
Thank you for your feedback. I have added AUTOTOOL-1156 to our internal product backlog to track this request.

**Comment by dstokesf5** [comment]
This was resolved with the 1.24.0 release of Declarative Onboarding.
